src/Castle.Windsor/MicroKernel/Resolvers/DefaultDependencyResolver.cs

```diff
@@ -265,8 +265,26 @@ private bool HasAnyComponentInValidState(Type service, DependencyModel dependenc
 			}
 
 			var handlers = kernel.GetHandlers(service);
-			return handlers.Where(handler => handler.IsBeingResolvedInContext(context) == false)
-				.Any(IsHandlerInValidState);
+			var nonResolvingHandlers = handlers.Where(handler => handler.IsBeingResolvedInContext(context) == false).ToList();
+			RebuildOpenGenericHandlersWithClosedGenericSubHandlers(service, context, nonResolvingHandlers);
```

Comment:
Do you think this change will have any impact on the `IGenericImplementationMatchingStrategy` extension point? Not sure how many unit tests that has.

Reply:
Hope not or else GenericImplementationWithGreaterArityThanServiceTestCase && GenericMultipleServicesComponentResolutionTestCase should be chucking up red flags. 

This is covered by tests that assert that these bits still work.